pytest/tests/mocknet/locust.py

```diff
@@ -0,0 +1,168 @@
+import argparse
+import cmd_utils
+import pathlib
+from rc import pmap, run
+import sys
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+import mocknet
+
+from configured_logger import logger
+
+
+def init_locust_node(instance_name):
+    node = mocknet.get_node(instance_name)
+    if node is None:
+        sys.exit(f'could not find node {instance_name}')
+    cmd_utils.init_node(node)
+    init_command = 'sudo apt update && sudo apt-get install -y git virtualenv build-essential python3-dev'
+    init_command += '&& git clone https://github.com/near/nearcore /home/ubuntu/nearcore '
+    init_command += '&& mkdir /home/ubuntu/locust'
+    init_command += '&& cd /home/ubuntu/locust && python3 -m virtualenv venv -p $(which python3) '
+    init_command += '&& ./venv/bin/pip install -r /home/ubuntu/nearcore/pytest/requirements.txt'
+    init_command += '&& ./venv/bin/pip install locust'
+    cmd_utils.run_cmd(node, init_command)
+
+
+def init_cmd(args):
+    nodes = [x for x in args.instance_names.split(',') if len(x) > 0]
+    pmap(init_locust_node, nodes)
+
+
+def parse_instsance_names(args):
+    if args.master is None:
+        print('master instance name?: ')
+        args.master = sys.stdin.readline().strip()
+
+    if args.workers is None:
+        print('''
+worker instance names? Give a comma separated list. It is also valid to
+have the machine used for the master process in this list as well:''')
+        args.workers = sys.stdin.readline().strip()
+
+    master = mocknet.get_node(args.master)
+    if master is None:
+        sys.exit(f'could not find node {args.master}')
+
+    worker_names = [x for x in args.workers.split(',') if len(x) > 0]
+    workers = [
+        mocknet.get_node(instance_name) for instance_name in worker_names
+    ]
+    for (name, node) in zip(worker_names, workers):
+        if node is None:
+            sys.exit(f'could not find node {name}')
+
+    return master, workers
+
+
+def upload_key(node, filename):
+    node.machine.upload(args.funding_key,
+                        '/home/ubuntu/locust/funding_key.json',
+                        switch_user='ubuntu')
+
+
+def run_master(args, node):
+    upload_key(node, args.funding_key)
+    cmd = f'/home/ubuntu/locust/venv/bin/python3 -m locust --web-port 3030 --master-bind-port 3000 -H {args.node_ip_port} -f locustfiles/{args.locustfile} --shard-layout-chain-id mainnet --funding-key=/home/ubuntu/locust/funding_key.json --max-workers {args.max_workers} --master'
+    cmd_utils.run_in_background(
+        node,
+        cmd,
+        'locust-master.txt',
+        pre_cmd=
+        'ulimit -S -n 100000 && cd /home/ubuntu/nearcore/pytest/tests/loadtest/locust'
+    )
+
+
+def wait_locust_inited(node, log_filename):
+    # We want to wait for the locust process to finish the initialization steps. Is there a better way than
+    # just waiting for the string "Starting Locust" to appear in the logs?
```

Comment:
Hm, unfortunately I don't know anything better. Maybe there is an API that you could query, but I couldn't find it after a quick search in the docs. I think this is actually fine.